### ISSUE TITLE
NTXentLoss, normalize issue.

### ISSUE THREAD

**Issue by jzhanghzau** [KevinMusgrave/pytorch-metric-learning#696]
Do we have to normalize the feature vector by ourself before sending these feature vectors into NTXentLoss. Because I checked the code in metric-learning, it seems that you don't include the normalize step.

Thanks in advance!

JJ

**Comment by jzhanghzau** [comment]
ah, I found you did the normalization step in calculating the similarity matrix.

another question is, assume I have a dataset, feature vector shape is [6, 100], label vector is [0,1,0,3,3,1]. Is it similar to supervised contrasitive learning if I use this NTXentloss in this case?

Thanks!

**Comment by KevinMusgrave** [comment]
If you're looking for [Supervised Contrastive Learning](https://arxiv.org/abs/2004.11362) there is a loss function for that: https://kevinmusgrave.github.io/pytorch-metric-learning/losses/#supconloss
